### ISSUE TITLE
Primary screen of fluid doc

### ISSUE THREAD

**Issue by shanyi15** [PaddlePaddle/Paddle#9275]
Related to #9031
@ranqiu92 ,please review this comment （they are written in 【】）
and add your review below each “【】”
Thanks!

### 基本使用概念 doc/fluid/getstarted/concepts
【保存模型】doc/fluid/getstarted/concepts/save_model
https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/getstarted/concepts/save_model/model_format.md
Design Doc: Model Format
【不放，篇幅短，此外，链接错误】

### 设计动机和目标 doc/fluid/design/motivation
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/motivation/api.md
PaddlePaddle Design Doc
【放，但需修复链接】
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/motivation/refactorization.md
Design Doc: Refactorization Overview
【放，但需修复链接】
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/motivation/fluid.md
Design Doc: PaddlePaddle Fluid
【放，但需修复一些小语法错误】
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/motivation/fluid_compiler.md
PaddlePaddle Fluid: Towards a Compiled Programming Language
【放，但需修复链接】

### 执行流程 doc/fluid/design/execution
1. 编译时流程
2. 运行时流程
3. 像写程序一样定义计算
(1) https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/execution/switch.md
Design Doc: Switch
【放】
(2) https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/execution/if_else_op.md
The IfElse Operator
【放】

### 核心概念  doc/fluid/design/concepts
1. 编译时概念  doc/fluid/design/concepts
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/concepts/README.md
Build system
【放】
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/concepts/cpp_data_feeding.md
C++ Data Feeding
【放】

2. 运行时概念 doc/fluid/design/concepts
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/concepts/functions_operators_layers.md
【放】
Design Doc: Functions, Operators, and Layers
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/concepts/program.md
【放，但需修复链接】
Design Doc: PaddlePaddle Programs
-  https://github.com/PaddlePaddle/Paddle/tree/develop/doc/fluid/design/concepts/variable.md
变量 
【放】
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/concepts/var_desc.md
var_desc.md variable描述
【放，需修复链接，加标题】
-  https://github.com/PaddlePaddle/Paddle/tree/develop/doc/fluid/design/concepts/tensor.md
张量 
【放】
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/concepts/tensor_array.md
Design for TensorArray
【放】
-  https://github.com/PaddlePaddle/Paddle/tree/develop/doc/fluid/design/concepts/lod_tensor.md
LOD 张量
【放】
-  https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/concepts/block.md
Design Doc: Block and Scope 
【放，但需修复链接】
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/concepts/scope.md
Design of Scope in Paddle
【放】
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/concepts/executor.md
Executor Design Doc
【放，但篇幅少需补充，且需修复链接】

### 数据类型 doc/fluid/design/data_type
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/data_type/float16.md
Design Doc: float16
【放】

### 内存管理 doc/fluid/design/memory
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/memory/memory_optimization.md
Memory Optimization 内存优化 
【放】
- https://github.com/PaddlePaddle/Paddle/tree/develop/doc/fluid/design/memory/README.md
基于区域的异构内存管理 是design doc
【感觉内容有点少，建议先不放】

### 多设备支持 doc/fluid/design/muti_devices
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/muti_devices/operator_kernel_type.md
Design Doc: The Keys of Operator Kernel Type
【放，需修复链接】
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/muti_devices/kernel_selection.md
kernel_selection
【放，但缺标题】 
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/muti_devices/kernel_hint_design.md
kernel_hint_design
【放，但需修复链接】

###  Dynamic RNN doc/fluid/design/dynamic_rnn
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/dynamic_rnn/rnn.md
RNNOp design
【放】
- https://github.com/PaddlePaddle/Paddle/tree/develop/doc/fluid/design/dynamic_rnn/rnn_design.md
RNN 变长输入设计
【放，ps..这是一篇中文的】

### 并发编程 doc/fluid/design/concurrent
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/concurrent/concurrent_programming.md
Design Doc: Concurrent Programming with Fluid
【放，修复链接】
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/concurrent/csp.md
Design Doc: CSP in PaddlePaddle Fluid
【不放，这篇看起来没写完】
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/concurrent/parallel_do.md
Design Doc: Parallel_Do in PaddlePaddle  属于多线程数据并行的设计文档
【放】

### 梯度更新算法 doc/fluid/design/algorithm
https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/algorithm/parameter_average.md
【放，但需修复链接】

### 复杂网络设计 doc/fluid/design/network
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/network/sequence_decoder.md
Design: Sequence Decoder Generating LoDTensors
【放，但需修复链接】
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/network/deep_speech_2.md
DeepSpeech2 on PaddlePaddle: Design Doc
【放】

### 代码结构和重要模块 doc/fluid/design/modules
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/modules/backward.md
Backward Building
【放】
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/modules/evaluator.md
Evaluator Design
【篇幅较少，有待商榷】
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/modules/python_api.md
Design Doc: Python API
【放，修复连接】
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/modules/regularization.md
Regularization in PaddlePaddle
【放，修复连接】
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/modules/infer_var_type.md
Design Doc: InferVarType
【放】
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/modules/optimizer.md
Optimizer Design
【放，修复链接】
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/modules/prune.md
Prune
【放】
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/modules/register_grad_op.md
Design Doc: Gradient Operators Registration
【放】
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/modules/selected_rows.md
Design Doc: Selected Rows
【放】
- https://github.com/PaddlePaddle/Paddle/tree/develop/doc/fluid/design/modules/net_op_design.md
Network Design
【放】
- https://github.com/PaddlePaddle/Paddle/tree/develop/doc/fluid/design/modules/batch_norm_op.md
批归一化

### fluid系统形态 doc/fluid/design/system
待定

### 多语言接口 doc/fluid/design/interface
-
 https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/interface/00.why_plain_c.md
Paddle多语言接口实现
【放】
-
 https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/interface/01.inference_implementation.md
C-API 模型推断实现文档
【放】

### 分布式训练 doc/fluid/design/dist_train
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/dist_train/distributed_architecture.md
Design Doc: Fluid Distributed Training Architecture
【放，但需修复连接】
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/dist_train/distributed_lookup_table_design.md
Design Doc: Distributed Lookup Table Operator
【放】
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/dist_train/multi_cpu.md
Design Doc: Execute the Program with Multi CPU
【不放】
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/dist_train/parameter_server.md
Design Doc: Parameter Server
【放】

### Others doc/fluid/design/others
-https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/others/gan_api.md
gan 这是一篇v2的文档，但此功能在v2没有实现，fluid可能会有这个功能，在写文档时可能会参考此文档
【不放】
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/others/graph_survey.md
Survey on Graph 一篇调研
【不放】
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/others/graph.md
Design Doc: Computations as a Graph 一篇不太完善的文档
【不放】
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/others/parameters_in_cpp.md
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/others/simple_op_design.md
【不放】
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/design/others/auto_gradient_check.md
Auto Gradient Check Design
【放】

### 性能分析与调优 doc/fluid/howto/performance
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/howto/performance/error_clip.md
Error Clip 作者给出翻译“误差截断”
【放】
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/howto/performance/profiler.md
profiler
【放，修复链接】

### 第三方库 doc/fluid/howto/third_party
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/howto/third_party/paddle_nccl.md
Design Doc: NCCL support in Paddle Fluid
【放，修复链接】

- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/howto/third_party/mkldnn_fluid.md
Design Doc: Add MKLDNN Kernel in Fluid Operator
【文档需要改，先不放】

## 开发标准 doc/fluid/dev
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/dev/support_new_device.md
Design Doc: Supporting new Device/Library
【放，改链接】
- https://github.com/PaddlePaddle/Paddle/blob/develop/doc/fluid/dev/releasing_process.md
PaddlePaddle发行规范 若放在官网里 可以命名为“如何发布版本”
【放】
- https://github.com/PaddlePaddle/Paddle/tree/develop/doc/fluid/dev/name_convention.md
operator参数命名规范
【放】
- https://github.com/PaddlePaddle/Paddle/tree/develop/doc/fluid/dev/op_markdown_format.md
Operators标准md格式
【放，修复链接】
